### DISCUSSION TITLE
How can I scan java files?

### DISCUSSION THREAD

**Discussion by huo2huo** [JetBrains/Qodana#84] [Q&A]
When I download the code and compile it.I can't scan the java file through qodana.
But when I open the compiled project in idea.After performing the following 2 steps, it can be scanned with qodana.
Step 1: execute 'add as Maven project'
![1](https://user-images.githubusercontent.com/101409437/165567411-87035c1b-f62a-4b43-9395-239386d0f713.png)

Step 1: select an SDK for the project
![2](https://user-images.githubusercontent.com/101409437/165567445-ff5e03e1-5ef3-466e-8dd2-20282d2da897.png)
My compiled statement is   'mvn clean -e install -P zpaas -Dmaven.test.skip=true -U'
![3](https://user-images.githubusercontent.com/101409437/165568230-a871794b-9c3a-4c08-ad47-bac64669206c.png)

I downloaded the code through Jenkins' pipeline and compiled it in Linux environment.
What should I do? In Linux environment, I can realize step 2 in idea

**Comment by 260947** [comment]
Did you solve it?
